libfreerdp/common/assistance.c

```diff
@@ -683,17 +673,17 @@ static BOOL freerdp_assistance_decrypt2(rdpAssistanceFile* file, const char* pas
 
 	cbOut += cbFinal;
 	cbFinal = 0;
-	pbOutW = (WCHAR*)pbOut;
-
-	if (cbOut > INT_MAX / 2)
-		goto fail;
 
-	cchOutW = (int)cbOut / 2;
-	file->ConnectionString2 = NULL;
-	status =
-	    ConvertFromUnicode(CP_UTF8, 0, pbOutW, cchOutW, &file->ConnectionString2, 0, NULL, NULL);
-
-	if (status <= 0)
+	union
+	{
+		const WCHAR* wc;
+		const BYTE* b;
+	} cnv;
```

Comment:
less compiler noise

Reply:
With this sample https://godbolt.org/z/TbsEsThes with `-pedantic-errors -Wall -Wextra` I wasn't able to produce any noise with any of the available compilers there (clang, gcc, msvc ...).

Maybe this happened with some ancient compilers but since we now rely on more up-to-date compilers I think these union hacks should be removed in favor of code clearness.

Reply:
ok, removed them all

Reply:
Still seeing these in `xf_rail.c`, `ncrypt_pkcs11.c`, `registry.c` and `stream.c` - maybe forgot to push?